Services/Models/InternalDeviceProperties.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models
+{
+    public interface IInternalDeviceProperties
+    {
+        Dictionary<string, object> GetAll();
+        void SetAll(Dictionary<string, object> newState);
+        bool Has(string key);
+        object Get(string key);
+        void Set(string key, object value);
+        bool Changed { get; }
+        void ResetChanged();
+    }
+
+    public class InternalDeviceProperties : IInternalDeviceProperties
+    {
+        private IDictionary<string, object> dictionary;
+        private bool changed;
+
+        public bool Changed { get { return this.changed; } }
+
+        public InternalDeviceProperties()
+        {
+            this.dictionary = new ConcurrentDictionary<string, object>();
+            this.changed = false;
+        }
+
+        public InternalDeviceProperties(DeviceModel deviceModel)
+        {
+            var intitalProperties = this.SetupProperties(deviceModel);
```

Comment:
typo

Reply:
to avoid code duplication and risk of diverging, you could refactor this method to extend the other ctor
```
public InternalDeviceProperties(DeviceModel deviceModel): this()
{
    this.dictionary = new ConcurrentDictionary<string, object>(
        this.SetupProperties(deviceModel));
}
```